### ISSUE TITLE
Add support for PyMdown Extensions: MagicLink and SaneHeaders, if feasible

### ISSUE THREAD

**Issue by kattni** [rvben/rumdl#355]
### Problem or motivation

[SaneHeaders](https://facelessuser.github.io/pymdown-extensions/extensions/saneheaders/) allows for starting a line with `#Content`, without rendering it as a header because it has no space before it. That specific feature isn't necessarily helpful if you're concerned about documentation contributors accidentally leaving the space out of actual headers.

[MagicLink](https://facelessuser.github.io/pymdown-extensions/extensions/magiclink/) enables you to provide something like `#10` in the content, and with appropriate configuration, it will auto-link to PR/Issue/Discussion 10 in the specified repository, among a number of other features.

When used together, it allows you to start a new paragraph with `#10` without rendering it as a header, and instead renders it as a link to the appropriate thing in the specified repo. Thing is, as you would be well aware, `rumdl` fails at this point with `[MD018] No space after # in heading`. It is worth noting that, as far as I can tell, there is no reason to add anything other than a numerical value after the `#` in this case.

### Proposed solution

Initially, I was wondering how this might be implemented without breaking the `MD018` rule entirely. A thought occurred as I was typing this out though.

Perhaps you could add a setting to `MD018` that allows you to inform `rumdl` that MagicLink is enabled, and at that point, it ignores `#nnn` where`nnn` is a numerical value, but not, for example, `#ActualHeaderContent`. 

I am not familiar enough with the `rumdl` implementation to know how to suggest anything useful, but that's what I came up with, in case it helps.

### Alternatives considered

Disable `rumdl` rule `MD018`inline each time this concept is present in Markdown.

### Feature category

Improve existing rule

### Example markdown

```markdown
# PRs that are helpful for context

#10 discusses the philosophy behind the project, and #37 shows a good example of how we expect code to be written.
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

### Additional context

This extension would be very helpful for including release notes in our documentation, as well as adding notes about issues to watch out for in certain situations. It would be excellent if `rumdl` had a way to work with it.

Thanks!

**Comment by rvben** [comment]
Thanks for raising this @kattni!

Implemented and will be in v0.1.7. Rather than a per-rule setting, we went with the existing flavor system — when using `flavor = "mkdocs"`, MD018 now skips MagicLink-style issue refs like `#10` at the start of lines, while still catching actual malformed headings like `#Summary`.

```toml
[global]
flavor = "mkdocs"
```

Does this work for your use case, or would you still prefer a dedicated per-rule setting?

**Comment by kattni** [comment]
This will work perfectly for me! Thank you so much. I do have a followup comment.

My thought is that MkDocs folks may not want to use MagicLink, and will then likely expect rumdl to fail if they insert a header like `#10 things you should know about this project`. So I'm wondering if, for that type of situation, that it might make more sense as a per-rule setting. Even I won't be using MagicLink on all of my MkDocs instances. I discussed losing that backup check from rumdl within my team, and agreed it was worth it to be able to use the GitHub shortlinks where applicable. But I feel like there will be folks who don't want to lose that check.

**Comment by rvben** [comment]
Thanks for the additional feedback. I have to think about this.

**Comment by rvben** [comment]
@kattni, I thought about it. I've now implemented MagicLink as a per-rule config option (breaking change), as the MagicLink is not part of the default mkdocs flavor, so enabling it by default was a mistake. You can configure it like this, regardless of flavor:

```toml
[MD018]
magiclink = true
```

When enabled, MD018 skips numeric patterns like `#10` at the start of lines (MagicLink-style issue/PR refs) while still catching actual malformed headings like #Summary.

When disabled (the default), all patterns are flagged — so `#10` things you should know would correctly trigger MD018.

This gives you precise control: enable it only on projects that use MagicLink, while keeping the full heading check on other MkDocs projects.

This will be in the next release.

**Comment by kattni** [comment]
@rvben Thank you so much for reconsidering! I deeply appreciate all the work you're doing here. It's been brilliant. rumdl has become an integral part of our workflow, and has made the move to MkDocs a huge improvement over the checks that were available for Sphinx and rST.

I have another project that it's not currently enabled on because of some bugs. I really need to go enable things and find all the issues and file them. I didn't have the bandwidth at the time, and it's gotten lost in the shuffle since. I'll definitely post them when I get to it.
